app/src/main/java/io/left/ripple/MainActivity.java

```diff
@@ -227,7 +227,13 @@ private void receiveColourMessage(RightMeshEvent rme) {
         int separatorIndex = dataString.indexOf(':');
 
         // Transmit the message forward if this device is not the intended final recipient.
-        MeshId recipient = new MeshId(dataString.substring(0, separatorIndex).getBytes());
+        MeshId recipient = null;
+        try {
+            recipient = MeshId.fromString(dataString.substring(0,separatorIndex));
+        } catch (RightMeshException e) {
```

Comment:
You might want to return here instead of continuing. Looks like the next line will NPE if it tries to call `recipient.equals()`.